### PR TITLE
RF: Make archive (de)compression use WitlessRunner

### PR DESCRIPTION
**Pull request by mih**
Another small step.

Went for a dedicated subshell to implement the pipe, rather than an exploration how to connect two runners. Not worse than what it was before.

### CODE REVIEW

File: datalad/support/archive_utils_7z.py [hunk 1]
```diff
@@ -52,13 +55,14 @@ def decompress_file(archive, dir_):
...
-        cmd = '7z x {} -so | 7z x -si -ttar'.format(quote_cmdlinearg(archive))
+        cmd = [
+            'sh', '-c',
+            '7z x {} -so | 7z x -si -ttar'.format(quote_cmdlinearg(archive)),
+        ]
     else:
         # fire and forget
         cmd = ['7z', 'x', archive]
-    runner.run(cmd)
+    runner.run(cmd, protocol=StdOutErrCapture)
```
Comment (kyleam) [inline datalad/support/archive_utils_7z.py:65]: Given that this function and the one below aren't returning the output, would it be better to use the `KillOutput` protocol?

Comment (mih) [reply datalad/support/archive_utils_7z.py:65]: Oh. You are absolutely correct!

Comment (mih) [reply datalad/support/archive_utils_7z.py]: Done

**Comment by mih** [comment]
Thanks for the review @kyleam. I made the switch to `KillOutput` and will merge now. All test failures are unrelated.

**Comment by bpoldrack** [comment]
Hm. `test_archives` just started failing in master in PY3.7: https://travis-ci.org/github/datalad/datalad/builds/725525494

I'm aware that it passed here, but since it only started *now*, seems related to this merge.

**Pull request merged**